### PR TITLE
add soft_fail per workload to buildkite.py

### PR DESCRIPTION
**Pull request by drgone**

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -27,6 +27,10 @@ steps:
       automatic:
         limit: {{.}}
   {{/retry}}
+  {{#softfail}}
+    soft_fail: true 
+        exit status: {{.}}
```
Comment (flaub) [inline ci/pipeline.yml]: drop this

File: ci/plan.yml [hunk 2]
```diff
@@ -198,6 +198,7 @@ SUITES:
         conda_env: conda/plaidml.yml
         precision: untested
         retry: 2
+        softfail: 1
```
Comment (flaub) [inline ci/plan.yml]: you could say, `soft_fail: yes`

**Comment by drgone** [comment]
@shiyao00 this adds mtl tests and allows soft fail on backend_test and mnist

**Comment by flaub** [comment]
I think we should constrain soft fail to a combination of platform + workflow. Allowing soft fail to happen for an entire platform means we won't notice when tests fail for platforms we know shouldn't be failing at all.

**Comment by drgone** [comment]
OK, this should work for any given workload/platform, use platform_overrides to soft_fail

platform_overrides:
  keras-stripe-mtl-uhd630:
    softfail: true

**Pull request merged**